core/templates/services/math-interactions.service.ts

```diff
@@ -374,6 +374,54 @@ export class MathInteractionsService {
 
     return partsList1.length === 0 && partsList2.length === 0;
   }
+
+  expressionMatchWithPlaceholders(
+      expressionWithPlaceholders: string, expressionWithoutPlaceholders: string,
+      placeholders: string[]): boolean {
+    // Check if expressionWithoutPlaceholders contains any placeholders.
+    for (let variable of nerdamer(expressionWithoutPlaceholders).variables()) {
+      if (placeholders.includes(variable)) {
+        return false;
+      }
+    }
+
+    // The expressions are first split into terms by addition and subtraction.
+    let termsWithPlaceholders = this.getTerms(expressionWithPlaceholders);
+    let termsWithoutPlaceholders = this.getTerms(expressionWithoutPlaceholders);
+
+    for (let i = termsWithPlaceholders.length - 1; i >= 0; i--) {
```

Comment:
Add comments for important sections in this loop as well.

Reply:
Done.